### PR TITLE
Spark: Add distribution mode not respected for CTAS/RTAS before 3.5.0

### LINKED ISSUE #8887: Document that distribution modes do not work with  DataFrameWriterV2 API  Create statements [Documentation]
### Apache Iceberg version

1.4.0 (latest release)

### Query engine

Spark

### Please describe the bug 🐞

Spark writes are currently using hash as default distribution modes, however distribution modes are not handeled in DataFrameWriterV2 API create statement.
There is a PR (https://github.com/apache/spark/pull/40734) that is supposed to handle this, but until then perhaps its a good idea to document this limitation and the requirement to manually deal with distribution in CTAS statements.
So this will not work as expected:
```scala
formattedDF
  .writeTo(fullTableName)
  .tableProperty("sort-order", "timestamp ASC NULLS FIRST")
  .tableProperty("write.distribution-mode", "range")
  .partitionedBy(days($"timestamp"))
  .create()
```
This might generate the following exception:
```
java.lang.IllegalStateException: Incoming records violate the writer assumption that records are clustered by spec and by partition within each spec. 
Either cluster the incoming records or switch to fanout writers.
```
Instead, distribution mode needs to be explicitly stated in create
```scala
  private def executeDDL(fullTablePath: String): DataFrame = {
    spark.sql(
      s"""
       |CREATE TABLE $fullTablePath (
       |  insert_timestamp TIMESTAMP,
       |  timestamp TIMESTAMP,
       |  role STRING,
       |  type STRING,
       |  file_key STRING,
       |  primary_dir STRING,
       |  secondary_dir STRING)
       |USING iceberg
       |PARTITIONED BY (days(timestamp))
       |TBLPROPERTIES (
       |  'format-version' = '2',
       |  'sort-order' = 'timestamp ASC NULLS FIRST',
       |  'write.distribution-mode' = 'range')
       |""".stripMargin
    )
    spark.sql(s"ALTER TABLE $fullTablePath WRITE ORDERED BY timestamp")
  }
```

### PR DESCRIPTION
**Pull request by manuzhang**
This fixes #8887

### CODE REVIEW

**Comment by manuzhang** [comment]
@nastra please help review.

File: docs/spark-writes.md [hunk 1]
```diff
@@ -343,7 +343,8 @@ data.writeTo("prod.db.sample").option("mergeSchema","true").append()
 Iceberg's default Spark writers require that the data in each spark task is clustered by partition values. This 
 distribution is required to minimize the number of file handles that are held open while writing. By default, starting
 in Iceberg 1.2.0, Iceberg also requests that Spark pre-sort data to be written to fit this distribution. The
-request to Spark is done through the table property `write.distribution-mode` with the value `hash`.
+request to Spark is done through the table property `write.distribution-mode` with the value `hash`. Spark doesn't respect
+distribution mode in CTAS/RTAS before 3.5.0.
```
Comment (nastra) [inline docs/spark-writes.md:347]: do we have tests that check the behavior in Spark 3.4 + Spark 3.5 on this?

Comment (manuzhang) [reply docs/spark-writes.md:347]: Tests added. Please check again.

File: spark/v3.4/spark/src/test/java/org/apache/iceberg/spark/sql/TestCreateTableAsSelect.java [hunk 2]
```diff
@@ -100,6 +101,15 @@ public void testPartitionedCTAS() {
         "Should have rows matching the source table",
         sql("SELECT * FROM %s ORDER BY id", sourceName),
         sql("SELECT * FROM %s ORDER BY id", tableName));
+
+    // Write distribution mode is not respected before Spark 3.5.0
+    Assert.assertThrows(
```
Comment (nastra) [inline spark/v3.4/spark/src/test/java/org/apache/iceberg/spark/sql/TestCreateTableAsSelect.java:106]: this should use `Assertions.assertThatThrownBy(() -> ...).isInstanceOf(SparkException.class).hasMessage(...)`

Comment (nastra) [reply spark/v3.4/spark/src/test/java/org/apache/iceberg/spark/sql/TestCreateTableAsSelect.java:106]: also it's probably better to just add a new test method: `ctasWriteDistributionModeNotRespected()`

File: docs/spark-writes.md (hunk 1 above)
Comment (nastra) [reply docs/spark-writes.md:347]: thanks. Just a few minor comments. Once those are resolved, this should be good to go

File: spark/v3.4/spark/src/test/java/org/apache/iceberg/spark/sql/TestCreateTableAsSelect.java [hunk 3]
```diff
@@ -102,6 +104,18 @@ public void testPartitionedCTAS() {
         sql("SELECT * FROM %s ORDER BY id", tableName));
   }
 
+  @TestTemplate
```
Comment (nastra) [inline spark/v3.4/spark/src/test/java/org/apache/iceberg/spark/sql/TestCreateTableAsSelect.java:107]:
```suggestion
  @Test
```

**Review by nastra — APPROVED** [review]

**Comment by nastra** [comment]
thanks @manuzhang for getting this done

**Pull request merged**